SimExLite/PMICalculators/SimpleScatteringPMICalculator.py

```diff
@@ -0,0 +1,11147 @@
+
+        #    self.g_s2e_pulse['sliceMax']   = self.g_s2e_pulse[''] - self.g_s2e_pulse['']
+        #    self.g_s2e_pulse['sliceMin']   = 0
+        #    self.g_s2e_pulse['sliceDelta']   = self.g_s2e_pulse['']  / ( self.g_s2e_pulse[''] - 1.0 )
+
+        #    self.g_s2e_pulse['slices']   =
+
+        #    self.g_s2e_pulse['xMax']   = ( self.g_s2e_pulse[''] - self.g_s2e_pulse[''] ) / 2.0
+        #    self.g_s2e_pulse['xMin']   = - self.g_s2e_pulse['']
+        #    self.g_s2e_pulse['xDelta']   = ( self.g_s2e_pulse[''] - self.g_s2e_pulse[''] ) / ( self.g_s2e_pulse[''] - 1.0 )
+
+        #    self.g_s2e_pulse['sel_pix_x']   = ( self.g_s2e_pulse[''] + 1 ) / 2
+        #    self.g_s2e_pulse['sel_pix_y']   = ( self.g_s2e_pulse[''] + 1 ) / 2
+
+        return
+
+    #    /**  Some transformation  and  derived useful parameters  */
+    #    data_prop_out.sliceMax = data_prop_out.sliceMax - data_prop_out.sliceMin ;
+    #    data_prop_out.sliceMin = 0 ;
+    #
+    #    data_prop_out.sliceDelta = data_prop_out.sliceMax / (double) (data_prop_out.nSlices-1) ;
+
+    #    data_prop_out.slices = calloc( data_prop_out.nSlices , sizeof(double) ) ;
+    #    for ( ii = 0 ; ii < data_prop_out.nSlices ; ii++ )
+    #    {
+    #        data_prop_out.slices[ii] = data_prop_out.sliceDelta * ii ;
+    #    }
+
+    #    data_prop_out.xMax =  ( data_prop_out.xMax - data_prop_out.xMin ) / 2.0 ;
+    #    data_prop_out.xMin = - data_prop_out.xMax ;
+    #    data_prop_out.xDelta = ( data_prop_out.xMax - data_prop_out.xMin ) /
+    #                           (double) (data_prop_out.nx - 1 ) ;
+
+    #    data_prop_out.yMax =  ( data_prop_out.yMax - data_prop_out.yMin ) / 2.0 ;
+    #
+    #    data_prop_out.yMin = - data_prop_out.yMax ;
+    #    data_prop_out.yDelta = ( data_prop_out.yMax - data_prop_out.yMin ) /
+    #                           (double) ( data_prop_out.ny - 1 ) ;
+
+    #    /**  Selecting center (ot highest fluence) pixel */
+    #    data_prop_out.sel_pix_x = ( data_prop_out.nx + 1 ) / 2 ;
+    #    data_prop_out.sel_pix_y = ( data_prop_out.ny + 1 ) / 2 ;
+
+    ##############################################################################
+
+    def f_time_evolution(self):
+
+        for step in range(1, self.g_s2e["steps"] + 1):
+            self.f_save_snp(step)
+
+    ##############################################################################
+
+
+def f_eval_disp(a_snp, a_r0, a_sample):
+
+    num_Z = len(list(a_sample["selZ"].keys()))
+    all_disp = numpy.zeros((num_Z,))
+    cc = 0
+    for sel_Z in list(a_sample["selZ"].keys()):
+        dr = a_snp["r"][a_sample["selZ"][sel_Z], :] - a_r0[a_sample["selZ"][sel_Z], :]
+        all_disp[cc] = numpy.mean(numpy.sqrt(numpy.sum(dr * dr, axis=1))) / 1e-10
+        cc = cc + 1
+    return all_disp
+
+
+def s2e_gen_randrot_quat(quat, rotmat):
+
+    if (
+        0
+        == quat[0] * quat[0] + quat[1] * quat[1] + quat[2] * quat[2] + quat[3] * quat[3]
```

Comment:
why not use **2?

Reply:
I think it was a habit from the C language, where multiplying is more efficient than using the "pow()" function. It's probably more sensible to use exponent now in python.